### PR TITLE
Development

### PR DESCRIPTION
**Pull request by queeksm**
Final merge due the completion of the program.

### CODE REVIEW

File: bubble_sort.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  swap = true
+  while swap
+    swap = false
+    (arr.length - 1).times do |x|
+      swap = true if arr[x] > arr[x + 1]
+      arr[x], arr[x + 1] = arr[x + 1], arr[x] if arr[x] > arr[x + 1]
+    end
+  end
+  print arr
+end
```
Comment (stratospherique) [inline bubble_sort.rb]: - **[REQUIRED]** You are supposed to **return arr**(the sorted array) not print it. :mag:

File: bubble_sort.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+bubble_sort_by(arr) { |a, b| a - b }
+
+unsorted = %w[hi hello hey]
+puts bubble_sort_by(unsorted) { |left, right| right.length - left.length }
+puts bubble_sort(unsorted)
```
Comment (stratospherique) [inline bubble_sort.rb]: - **[REQUIRED]** No debugging instructions in committed code (as discussed [here](https://questions.microverse.org/t/debugging-instructions-code/346)) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/section/debugging_requirement.png) from Pathwright)

File: bubble_sort.rb [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+      swap = true if arr[x] > arr[x + 1]
+      arr[x], arr[x + 1] = arr[x + 1], arr[x] if arr[x] > arr[x + 1]
+    end
+  end
+  print arr
+end
+
+arr = [1, 4, 5, 66, 7, 3, 12_312, 2]
+
+bubble_sort(arr)
```
Comment (stratospherique) [inline bubble_sort.rb]: - **[REQUIRED]** No debugging instructions in committed code (as discussed [here](https://questions.microverse.org/t/debugging-instructions-code/346)) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/section/debugging_requirement.png) from Pathwright)

File: bubble_sort.rb [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  while swap
+    swap = false
+    (arr.length - 1).times do |x|
+      b = arr[x + 1]
+      arr[x], arr[x + 1] = arr[x + 1], arr[x] if yield(arr[x], b) >= 0
+      swap = true if yield(arr[x], b) >= 0
+    end
+  end
+  print arr
+end
```
Comment (stratospherique) [inline bubble_sort.rb]: - **[REQUIRED]** You are supposed to **return arr**(the sorted array) not print it. :mag:

**Review by stratospherique — CHANGES_REQUESTED** [review]
### Great start @team :clap: However, some changes needs to be applied in order to get your methods working and the project approved. check :arrow_down: 

- **[REQUIRED]** Check the code comments below, they should help you refactor you methods.

### That's it :relaxed: When finished, kindly resubmit another code review request :pray: 

Happy coding :v: 
**_Ahmed Mahfoudh_**

File: bubble_sort.rb (hunk 2 above)
Comment (Rhelli) [reply bubble_sort.rb]: Debugging instructions on lines 34-38 have been removed (as well as on line 17). Thanks!

File: bubble_sort.rb (hunk 4 above)
Comment (Rhelli) [reply bubble_sort.rb]: I've changed *print arr* to just *arr* for an implicit return!

**Review by imhta — APPROVED** [review]
# Project approved 👍 

### Hello  @Rhelli @queeksm, 👋 You have done a good job 👏 



----
**Pon Muthu Selvam N**

**Comment by Rhelli** [comment]
> # Project approved 👍
> ### Hello @Rhelli @queeksm, 👋 You have done a good job 👏
> 
> 
> **Pon Muthu Selvam N**

Xrsa3jd

@imhta @stratospherique Thank you very much! 👌

**Pull request merged**